public/js/updateBook.ts

```diff
@@ -9,52 +9,482 @@ interface Book {
+            inputBox.addEventListener("input", () => {
+                inputBox.setCustomValidity("");
+                inputBox.reportValidity();
+            });
+
+            inputBox.addEventListener("blur", function () {
+                validateAuthorBox(inputBox);
+            })
+        }
+    });
+
+    inputGenreBoxes.forEach((inputBox) => {
+        if (inputBox instanceof HTMLInputElement) {
+            inputBox.addEventListener("input", () => {
+                inputBox.setCustomValidity("");
+                inputBox.reportValidity();
+            });
+
+            inputBox.addEventListener("blur", function () {
+                validateGenreBox(inputBox);
+            })
+        }
+    });
+
+    publisherInput.addEventListener("input", function () {
+        publisherInput.setCustomValidity("");
+    });
+
+    publisherInput.addEventListener("blur", function () {
+        validatePublisher();
+    });
+
+    publishedInput.addEventListener("input", function () {
+        publishedInput.setCustomValidity("");
+    });
+
+    publishedInput.addEventListener("blur", function () {
+        validatePublished();
+    });
+
+    coverUrlInput.addEventListener("input", function () {
+        coverUrlInput.setCustomValidity("");
+    });
+
+    coverUrlInput.addEventListener("blur", function () {
+        validateCoverUrl();
+    });
+
+    const initialAddAuthorBtn = document.getElementById("addAuthor") as HTMLButtonElement;
+    initialAddAuthorBtn.addEventListener("click", function () {
+        createInputAuthorBox(0);
+    });
+
+    const initialAddGenreBtn = document.getElementById("addGenre") as HTMLButtonElement;
+    initialAddGenreBtn.addEventListener("click", function () {
+        createInputGenereBox(0);
+    });
+
+    const submitBtn = document.getElementById("submitUpdateBookForm") as HTMLButtonElement;
+    submitBtn.addEventListener("click", async (event) => {
+        event.preventDefault();
+        submitUpdateBook();
+    });
+
+    async function updateBookCall(bookTitle: string, authors: string[], genres: string[], publisher: string, published: string, coverUrl: string, abstract: string, admin: string) {
+        const requestBody = {
+            title: bookTitle,
+            isbn: isbn,
+            abstract: abstract,
+            image: coverUrl,
+            authors,
+            genres,
+            publisher: publisher,
+            admin: admin,
+            datePublished: published,
+        };        
+
+        try {
+            const response = await fetch('/book', {
+                method: 'PUT',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(requestBody),
+            });
+
+            if (response.ok) {
+                return true;
+            } else {
+                return false;
+            }
+        } catch (error) {
+            return false;
```

Comment:
I dont think it is a good idea to just return false when an error is caught. We should throw it and display the message on the frontend. 
Same with the `else` in the `try`.

Reply:
That will return the boolean which is handled by showing a success popup or a failure pop up